R/appenders.R

```diff
@@ -89,7 +89,7 @@ appender_file <- function(file, append = TRUE, max_lines = Inf, max_bytes = Inf,
                 n_lines <- tryCatch(
                     suppressWarnings(R.utils::countLines(file)),
                     error = function(e) 0)
-                n_bytes <- ifelse(file.exists(file), file.info(file)$size, 0)
+                n_bytes <- if (file.exists(file)) file.info(file)$size else 0
```

Comment:
what's the advantage of calling `if/else` (without the curly braces) compared to `ifelse`?

Reply:
It is more secure and readable. 
https://twitter.com/hadleywickham/status/928983687605059585?lang=bn

Reply:
It is more secure and readable.
Here a Hadley Wickham opinion who is in favour of such step.
https://twitter.com/hadleywickham/status/928983687605059585

Reply:
Strongly agree with @Polkas . Note also this edge case (not relevant for `file.exists(file)`, since it can not return `NA`, but still):
```
> ifelse(NA, TRUE, FALSE)
[1] NA
> `if`(NA, TRUE, FALSE)
Error in if (NA) TRUE else FALSE : missing value where TRUE/FALSE needed
```
Note also that for simple conditionals, I tend to use backticked `if` to keep the functional style.

Reply:
Hm, got it, thanks for the pointers. Although makes sense from a perspective, but I have an aversion to using `if` without the curly braces, as have been hurt by how the parser handles it (single VS multiple statements), e.g. with `covr` false positives or missed `else` branches on multiline expressions.

Although I've not run into this issue recently, habits die hard :) 

And the R manual of `if` still suggests always keeping the braces .. which makes it ugly for single-line statements.

Anyway, I will keep thinking about it, and looking at the rest of the PR in the meanwhile -- thanks again!!

Reply:
I understand your doubts, so what about `ifelse(isTRUE(cond), ..., ...)`.

Reply:
Sorry, let's not wrap `file.exists` in `isTRUE` :zipper_mouth_face: 

I don't think there's any benefit in that extra check, still makes the code more complex and a bit slower.

So far, the backtick + `if` combo is the most appealing to me .. but still feels like black magic, so probably we need something easier to grasp -- which might be just the ugly multi-line `if` conditionals?

Anyway, this is more of a language syntax question, and I feel like we just miss a nice R implementation of the ternary operator, which is unlikely that we will resolve in this thread :) 

So I might be open to updating the above to avoid `if` conditionals altogether with something like:

```r
na.omit(c(file.info(file)$size, 0))[1]
```

But let me rather check the rest of the PR instead at last :)